velox/functions/lib/window/tests/WindowTestBase.cpp

```diff
@@ -48,6 +50,51 @@ WindowTestBase::QueryInfo WindowTestBase::buildWindowQuery(
   return {op, functionSql, querySql};
 }
 
+WindowTestBase::QueryInfo WindowTestBase::buildStreamingWindowQuery(
+    const std::vector<RowVectorPtr>& input,
+    const std::string& function,
+    const std::string& overClause,
+    const std::string& frameClause) {
+  std::string functionSql =
+      fmt::format("{} over ({} {})", function, overClause, frameClause);
+  // split the order by following sql into the orderBy({})
+  core::PlanNodePtr op = nullptr;
+  std::string orderStr = "order by";
+  auto startIdx = overClause.find("order by");
+  if (startIdx != std::string::npos) {
+    // find order by
+    auto endIdx = startIdx + orderStr.length() - 1;
+    auto overClauseLength = overClause.length();
+    auto orderByPars = overClause.substr(endIdx + 1, overClauseLength);
+    std::istringstream tokenStream(orderByPars);
+    std::string token;
+    std::vector<std::string> orderByParsVec;
+    while (std::getline(tokenStream, token, ',')) {
+      orderByParsVec.push_back(token);
+    }
+    op = PlanBuilder()
```

Comment:
In order for StreamingWindow to have correct behavior the sorting keys for the ORDER BY node here have to be Window partition keys followed by order_by keys. 

The current behavior only works if you have 1 row value per partition not more.

Reply:
Thank you very much for pointing out this issue. Updated.